src/main/java/com/example/springbootdocker/utils/PageResponseUtils.java

```diff
@@ -0,0 +1,36 @@
+package com.example.springbootdocker.utils;
+
+import com.example.springbootdocker.model.Category;
+import com.example.springbootdocker.model.Product;
+import com.example.springbootdocker.response.PageResponse;
+import lombok.experimental.UtilityClass;
+import org.springframework.data.domain.Page;
+
+@UtilityClass
+public class PageResponseUtils {
+
+    public PageResponse<Category> getCategoryPageResponse(String direction, String sortBy, Page<Category> categoryPage) {
+        return PageResponse.<Category>builder()
+                .content(categoryPage.getContent())
+                .size(categoryPage.getSize())
+                .page(categoryPage.getNumber())
+                .sortBy(sortBy)
+                .direction(direction)
+                .totalElements(categoryPage.getTotalElements())
+                .totalPages(categoryPage.getTotalPages())
+                .build();
+    }
+
+    public PageResponse<Product> getProductPageResponse(String direction, String sortBy, Page<Product> productPage) {
```

Comment:
Hi, why can't you write a single parametrized method instead of these both?

    public <T> PageResponse<T> getProductPageResponse(String direction, String sortBy, Page<T> page) {
        return PageResponse.<T>builder()
                .content(page.getContent())
                .size(page.getSize())
                .page(page.getNumber())
                .sortBy(sortBy)
                .direction(direction)
                .totalElements(page.getTotalElements())
                .totalPages(page.getTotalPages())
                .build();
    }

Reply:
Hi, thank you for discussion, just made it fast and didn't think about parameterizing , but yeah, of course it's much better so I pushed changes